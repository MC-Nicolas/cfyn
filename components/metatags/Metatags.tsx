import Head from 'next/head';

type Props = {
  title: string;
  description?: string;
  image?: string;
};

const Metatags = ({ title, description, image }: Props) => {
  return (
    <Head>
      <title>{title}</title>
      <meta name='viewport' content='width=device-width, initial-scale=1.0' />
      <meta name='twitter:card' content='summary' />
      <meta name='twitter:site' content='@CFYN' />
      <meta name='twitter:title' content={title} />
      <meta name='twitter:description' content={description} />
      <meta name='twitter:image' content={image} />

      <meta property='og:title' content={title} />
      <meta property='og:description' content={description} />
      <meta property='og:image' content={image} />
    </Head>
  );
};
export default Metatags;
