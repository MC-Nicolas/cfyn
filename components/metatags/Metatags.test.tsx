/**
 * @jest-environment jsdom
 */

import React from 'react';
import {
  render,
  cleanup,
  waitFor,
  waitForDomChange,
} from '@testing-library/react';
import Metatags from './Metatags';

const meta = {
  title: 'CFYN',
  description: 'Your finance made easy !',
};
jest.mock('next/head', () => {
  return {
    __esModule: true,
    default: ({ children }: { children: Array<React.ReactElement> }) => {
      return <>{children}</>;
    },
  };
});

test('<Metatags />', async () => {
  const wrapper = render(
    <Metatags title={meta.title} description={meta.description} />,
    { container: document.head }
  );
  await waitFor(() => {
    expect(document.title).toBe('CFYN');
    expect(wrapper).toMatchSnapshot();
  });
});
