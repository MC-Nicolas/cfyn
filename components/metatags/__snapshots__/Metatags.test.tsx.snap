// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Metatags /> 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <head>
    <title>
      CFYN
    </title>
    <meta
      content="width=device-width, initial-scale=1.0"
      name="viewport"
    />
    <meta
      content="summary"
      name="twitter:card"
    />
    <meta
      content="@CFYN"
      name="twitter:site"
    />
    <meta
      content="CFYN"
      name="twitter:title"
    />
    <meta
      content="Your finance made easy !"
      name="twitter:description"
    />
    <meta
      name="twitter:image"
    />
    <meta
      content="CFYN"
      property="og:title"
    />
    <meta
      content="Your finance made easy !"
      property="og:description"
    />
    <meta
      property="og:image"
    />
  </head>,
  "container": <head>
    <title>
      CFYN
    </title>
    <meta
      content="width=device-width, initial-scale=1.0"
      name="viewport"
    />
    <meta
      content="summary"
      name="twitter:card"
    />
    <meta
      content="@CFYN"
      name="twitter:site"
    />
    <meta
      content="CFYN"
      name="twitter:title"
    />
    <meta
      content="Your finance made easy !"
      name="twitter:description"
    />
    <meta
      name="twitter:image"
    />
    <meta
      content="CFYN"
      property="og:title"
    />
    <meta
      content="Your finance made easy !"
      property="og:description"
    />
    <meta
      property="og:image"
    />
  </head>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
